testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/fragment/Sidebar.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.ui.account2.page.fragment;
+
+import org.jboss.arquillian.graphene.fragment.Root;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.keycloak.testsuite.util.UIUtils.clickLink;
+import static org.keycloak.testsuite.util.WaitUtils.pause;
+
+/**
+ * @author Vaclav Muzikar <vmuzikar@redhat.com>
+ */
+public class Sidebar extends AbstractFragmentWithMobileLayout {
+    public static int MOBILE_WIDTH = 767; // if the page width is less or equal than this, we expect the sidebar to be collapsed by default
+    public static final String NAV_ITEM_ID_PREFIX = "nav-link-";
+
+    @Root
+    private WebElement sidebarRoot;
+    @FindBy(id = "nav-toggle") // relative to root element
+    private WebElement collapseToggle;
+
+    @Override
+    protected int getMobileWidth() {
+        return MOBILE_WIDTH;
+    }
+
+    public boolean isCollapsed() {
+        return sidebarRoot.getAttribute("class").contains("collapsed");
+    }
+
+    public void collapse() {
+        assertFalse("Sidebar is already collapsed", isCollapsed());
+        collapseToggle.click();
+        pause(2000); // wait for animation
+        assertTrue("Sidebar is not collapsed", isCollapsed());
+    }
+
+    public void expand() {
+        assertTrue("Sidebar is already expanded", isCollapsed());
+        collapseToggle.click();
+        pause(2000); // wait for animation
```

Comment:
From my experience, I don't think we can use a dynamic wait for animations. It was always a bit unstable and in cases like this it's even more problematic and even unusable. When an animations leads to element being displayed (fade in, expand menu, ...), you cannot check if the element is present as it actually is present (and visible) for the whole duration of the animation. 

Of course there might be some other way how to check if an animation is finished. But it be quite complicated mechanism specific for given animation and complicated usually equals unstable in case of some browsers (bear in mind we will be running tests with IE and mobile browsers as well).

Luckily for us, animations take a predefined amount of time (unlike e.g. loading a page) – so there's really no harm in statically waiting for an animation.

But feel free to create a JIRA for this. We might investigate in the future how to approach animations in a different way.